### ISSUE TITLE
MAPL3 Unit Tests failing with GNU macOS

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/MAPL#2563]
A test build of MAPL3 (or rather the branch in #2558) today on macOS found unit test failures. Namely `MAPL.generic3g.tests` and *almost* all of the extdata2g tests fail:

```
The following tests FAILED:
	 11 - MAPL.generic3g.tests (Failed)
	 42 - ExtData2G_case1 (Failed)
	 43 - ExtData2G_case2 (Failed)
	 44 - ExtData2G_case3 (Failed)
...
	 69 - ExtData2G_case28 (Failed)
	 70 - ExtData2G_case29 (Failed)
Errors while running CTest
```

Only case19 works, but that seems to be the "do nothing" test:
```
  19. Test set file to /dev/null
```

This seems so comprehensive that maybe fixing the generic3g test would then fix the extdata2g?

**Comment by mathomp4** [comment]
It looks like the common error is:
```
 Start: <Test_SimpleParentGridComp_suite.test_child_user_items_created[npes=0][npes=0]>
. ERROR AT: /Users/mathomp4/Models/MAPL3-TomPR-2558-GNU/MAPL/build-Debug/generic3g/tests/Test_SimpleParentGridComp.F90          41         545
 ERROR AT: /Users/mathomp4/Models/MAPL3-TomPR-2558-GNU/MAPL/build-Debug/generic3g/tests/Test_SimpleParentGridComp.F90          41         545
 ERROR AT: /Users/mathomp4/Models/MAPL3-TomPR-2558-GNU/MAPL/build-Debug/generic3g/tests/Test_SimpleParentGridComp.F90          41         545
 ERROR AT: /Users/mathomp4/Models/MAPL3-TomPR-2558-GNU/MAPL/build-Debug/generic3g/tests/Test_SimpleParentGridComp.F90          41         545
 Failure in <Test_SimpleParentGridComp_suite.test_child_user_items_created[npes=0][npes=0]>
F   end: <Test_SimpleParentGridComp_suite.test_child_user_items_created[npes=0][npes=0]>
```

I believe that points to:
```fortran
   call NUOPC_FieldDictionarySetup('./scenarios/FieldDictionary.yml', _RC)
```

**Comment by bena-nasa** [comment]
I will take a look

**Comment by tclune** [comment]
The generic tests failures are almost certainly unrelated.     Unfortunately, NAG tests pass on laptop, so that's all I've been running.

**Comment by tclune** [comment]
Hmm.  The NUOPC dictionary mechanism is not really being used yet - just a placeholder. And that bit has not changed in a while.

**Comment by bena-nasa** [comment]
@tclune @mathomp4 
I can't reproduce what @mathomp4 said failed on macOS with regards to ExtData, here is what failed for me, on 2 of the extdata tests failed and those are the ones that need a lot of cores:
```
	  1 - MAPL.pfio.tests (Failed)
	  4 - pFIO_tests_hybrid (Failed)
	  8 - MAPL.profiler.tests (Failed)
	  9 - MAPL.generic.tests (Failed)
	 10 - MAPL.generic3g.tests (Failed)
	 11 - MAPL.field_utils.tests (Failed)
	 13 - MAPL.base.tests (Failed)
	 16 - MAPL.geom_mgr.tests (Failed)
	 17 - MAPL.regridder_mgr.tests (Failed)
	 28 - ExtData1G_case12 (Failed)
	 52 - ExtData2G_case12 (Failed)
```
This is with gcc 12.2 and openmpi 5.0.0

**Comment by bena-nasa** [comment]
Ok, now if I use this branch
feature/tclune/mapl3-udunits-refactor
this is what fails

```
	  1 - udunits2f.tests (Failed)
	 11 - MAPL.generic3g.tests (Failed)
	 29 - ExtData1G_case12 (Failed)
	 53 - ExtData2G_case12 (Failed)
```

**Comment by mathomp4** [comment]
> Ok, now if I use this branch feature/tclune/mapl3-udunits-refactor this is what fails
> 
> ```
> 	  1 - udunits2f.tests (Failed)
> 	 11 - MAPL.generic3g.tests (Failed)
> 	 29 - ExtData1G_case12 (Failed)
> 	 53 - ExtData2G_case12 (Failed)
> ```

I'll try again from scratch tomorrow. My clone might be all out of whack.

**Comment by mathomp4** [comment]
Ugh. So this is fun. So I tried a couple builds of the @tclune branch. In one case I used my "gcc" environment which is gcc-12/g++-12/gfortran-12 all from brew. In another I used my "clanggfortran" environment which is clang/clang++/gfortran-12.

With all gcc:
```
55% tests passed, 28 tests failed out of 62
...
```
as seen above. With clang-gfortran:
```
100% tests passed, 0 tests failed out of 62
```

So now we have *three* results. Ben failed generic3g and case12 (I think I know why the udunits2f test failed). I have one that fails everything and one that passes all!

I'm going to re build the baselibs for my gcc env to make sure maybe nothing went crazy on my end.
